bokeh/models/plots.py

```diff
@@ -383,18 +384,18 @@ def _check_missing_renderers(self):
     @error(BAD_EXTRA_RANGE_NAME)
     def _check_bad_extra_range_name(self):
         msg  = ""
-        filt = lambda x: x is not self and isinstance(x, Plot)
-        for ref in collect_filtered_models(filt, self):
-            prop_names = ref.properties()
-            bad = []
-            if 'x_range_name' in prop_names and 'y_range_name' in prop_names:
-                if ref.x_range_name not in self.extra_x_ranges and ref.x_range_name != "default":
-                    bad.append(('x_range_name', ref.x_range_name))
-                if ref.y_range_name not in self.extra_y_ranges and ref.y_range_name != "default":
-                    bad.append(('y_range_name', ref.y_range_name))
-            if bad:
-                if msg: msg += ", "
-                msg += (", ".join("%s=%r" % (a, b) for (a,b) in bad) + " [%s]" % ref)
+        for place in _VALID_PLACES + ('renderers',):
+            for ref in getattr(self, place):
+                prop_names = ref.properties()
+                bad = []
+                if 'x_range_name' in prop_names and 'y_range_name' in prop_names:
+                    if ref.x_range_name not in self.extra_x_ranges and ref.x_range_name != "default":
+                        bad.append(('x_range_name', ref.x_range_name))
+                    if ref.y_range_name not in self.extra_y_ranges and ref.y_range_name != "default":
+                        bad.append(('y_range_name', ref.y_range_name))
```

Comment:
As long as we are already making changes here: I do think that the existing logic here admits the possibility of an error if some day there is a ref with *one* of `x_range_name` or `y_range_name`  but not  both. @poldavezac I won't insist on any changes but just in case you want to try to tidy up and improve the logic here I'll wait to merge until later today.